From 6202669b8e8f50837448e2d4c482671abf72f648 Mon Sep 17 00:00:00 2001
From: Iru Cai <mytbk920423@gmail.com>
Date: Sun, 5 Nov 2023 17:12:43 +0800
Subject: [PATCH 1/1] mb/hp/elitebook_820_g2: do not set EC SLPT on S5

Setting EC SLPT bit in S5 will make HP EliteBook 820 G2
fail to reboot under Linux 6.1 and later kernel versions.

Change-Id: I48f5a35cd78db3b32d9f76cb8e266c738da34e7c
Signed-off-by: Iru Cai <mytbk920423@gmail.com>
Reviewed-on: https://review.coreboot.org/c/coreboot/+/78907
Tested-by: build bot (Jenkins) <no-reply@coreboot.org>
Reviewed-by: Felix Singer <service+coreboot-gerrit@felixsinger.de>
---
 src/mainboard/hp/elitebook_820_g2/acpi/platform.asl | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/mainboard/hp/elitebook_820_g2/acpi/platform.asl b/src/mainboard/hp/elitebook_820_g2/acpi/platform.asl
index 8023ae826c..d17f575c40 100644
--- a/src/mainboard/hp/elitebook_820_g2/acpi/platform.asl
+++ b/src/mainboard/hp/elitebook_820_g2/acpi/platform.asl
@@ -10,5 +10,8 @@ Method(_WAK,1)
 
 Method(_PTS,1)
 {
-	\_SB.PCI0.LPCB.EC0.SLPT = Arg0
+	If (Arg0 != 5)
+	{
+		\_SB.PCI0.LPCB.EC0.SLPT = Arg0
+	}
 }
-- 
2.39.2

